### PR TITLE
[component: agw]: Periodic Registration fixes

### PR DESCRIPTION
**Pull request by panyogesh**
Fixes:
  1. Support for Periodic Registration
  2. Support for timer 3550
  3. Fixes for GUTI encoding/decoding
  4. Fixes for Context lookup based on GUTI
  5. Minor Clean ups
  6. NGKSI Fixes

Test Covered :
  1. Periodic Registration Idle Mode/Active Mode
  2. All P0 related testcases
  3. Basic Registration/De-Registration in loop
  4. SIT issue for ngKSI
  5. Validated in TerraVM and UERANSIM setup

Signed-off-by: Yogesh Pandey <yogesh@wavelabs.ai>

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/c/oai/tasks/amf/amf_recv.cpp [hunk 1]
```diff
@@ -37,37 +37,25 @@ int amf_handle_service_request(
...
-  OAILOG_INFO(LOG_NAS_AMF, " TMSI received %08" PRIx32 "\n", tmsi_rcv);
 
-  if (tmsi_rcv == tmsi_stored) {
+  if (ue_context) {
     OAILOG_INFO(
         LOG_NAS_AMF,
         "TMSI matched for the UE id %d "
-        " receved TMSI %08X stored TMSI %08X \n",
-        ue_id, tmsi_rcv, tmsi_stored);
+        " receved TMSI %08X\n",
```
Comment (ulaskozat) [inline lte/gateway/c/oai/tasks/amf/amf_recv.cpp]: typo "receved"

File: lte/gateway/c/oai/tasks/amf/Registration.cpp [hunk 2]
```diff
@@ -800,33 +785,25 @@ int amf_proc_registration_complete(
...
-      OAILOG_DEBUG(
-          LOG_NAS_AMF,
-          "UE " AMF_UE_NGAP_ID_FMT
-          " REGISTRATION COMPLETE discarded (AMF procedure not found)\n",
-          ue_id);
-      bdestroy((bstring)(smf_msg_pP));
     }
   } else {
-    OAILOG_WARNING(LOG_NAS_AMF, "UE Context not found..\n");
+    OAILOG_WARNING(LOG_NAS_AMF, "UE Context not found...ue_id=%d\n", ue_id);
```
Comment (ulaskozat) [inline lte/gateway/c/oai/tasks/amf/Registration.cpp]: utilize AMF_UE_NGAP_ID_FMT rather than `%d`.

File: lte/gateway/c/oai/tasks/amf/amf_as.cpp [hunk 3]
```diff
@@ -422,6 +438,9 @@ static int amf_reg_acceptmsg(
   offset++;
   nas_msg->security_protected.plain.amf.msg.registrationacceptmsg.mobile_id
       .mobile_identity.guti.tmsi3 = *offset;
+  offset++;
+  nas_msg->security_protected.plain.amf.msg.registrationacceptmsg.mobile_id
+      .mobile_identity.guti.tmsi4 = *offset;
 
   // TAI List, Allowed NSSAI and GPRS Timer 3 harcoded
```
Comment (ulaskozat) [inline lte/gateway/c/oai/tasks/amf/amf_as.cpp]: typo `harcoded`

File: lte/gateway/c/oai/tasks/amf/Registration.cpp (hunk 2 above)
Comment (panyogesh) [reply lte/gateway/c/oai/tasks/amf/Registration.cpp]: Comments addressed. Fixed same in other parts of PR as well.

File: lte/gateway/c/oai/tasks/amf/amf_as.cpp (hunk 3 above)
Comment (panyogesh) [reply lte/gateway/c/oai/tasks/amf/amf_as.cpp]: Comments addressed.

**Pull request merged**